python_modules/dagster/dagster_tests/asset_defs_tests/test_assets_job.py

```diff
@@ -1992,6 +1993,46 @@ def unpartitioned_asset():
         frozenset(["daily_asset_different_start_date", "unpartitioned_asset"]),
     }
 
+@ignore_warning("Function `observable_source_asset` is experimental")
+def test_get_base_asset_jobs_multiple_partitions_defs_and_observable_assets():
+    class B:
+        ...
+
+    class X:
```

Comment:
Is this class definition necessary?

Reply:
Yes! without this, for example if using int, the bug would not be triggered. In that case, the inputs that cause the problem, happen to have a type loader (not sure why). Then The check [here](https://github.com/dagster-io/dagster/blob/044c2cb1729a8dbc27fdfc37a0543be69b9b4742/python_modules/dagster/dagster/_core/definitions/op_definition.py#L334) skips those inputs, and the problem does not manifest.

Reply:
Ah, you are right, X is not used. only B is used. Sorry.